src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -0,0 +1,246 @@
+                        queries.add(parseInnerQueryBuilder(parser));
+                        token = parser.nextToken();
+                    }
+                } else {
+                    throw new ParsingException(
+                        parser.getTokenLocation(),
+                        String.format(Locale.ROOT, "[%s] query does not support [%s]", NAME, currentFieldName)
+                    );
+                }
+            } else {
+                if (AbstractQueryBuilder.BOOST_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    boost = parser.floatValue();
+                } else if (AbstractQueryBuilder.NAME_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    queryName = parser.text();
+                } else {
+                    throw new ParsingException(
+                        parser.getTokenLocation(),
+                        String.format(Locale.ROOT, "[%s] query does not support [%s]", NAME, currentFieldName)
+                    );
+                }
+            }
+        }
+
+        if (!queriesFound) {
+            throw new ParsingException(
+                parser.getTokenLocation(),
+                String.format(Locale.ROOT, "[%s] requires 'queries' field with at least one clause", NAME)
+            );
+        }
+
+        HybridQueryBuilder compoundQueryBuilder = new HybridQueryBuilder();
+        compoundQueryBuilder.queryName(queryName);
+        compoundQueryBuilder.boost(boost);
+        for (QueryBuilder query : queries) {
+            compoundQueryBuilder.add(query);
+        }
+        return compoundQueryBuilder;
+    }
+
+    protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws IOException {
+        HybridQueryBuilder newBuilder = new HybridQueryBuilder();
+        boolean changed = false;
+        for (QueryBuilder query : queries) {
+            QueryBuilder result = query.rewrite(queryShardContext);
+            if (result != query) {
+                changed = true;
+            }
+            newBuilder.add(result);
+        }
+        if (changed) {
+            newBuilder.queryName(queryName);
+            newBuilder.boost(boost);
+            return newBuilder;
+        } else {
+            return this;
+        }
+    }
+
+    @Override
+    protected boolean doEquals(HybridQueryBuilder obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+        EqualsBuilder equalsBuilder = new EqualsBuilder();
+        equalsBuilder.append(fieldName, obj.fieldName);
+        equalsBuilder.append(queries, obj.queries);
+        return equalsBuilder.isEquals();
+    }
+
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(queries);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    static List<QueryBuilder> readQueries(StreamInput in) throws IOException {
```

Comment:
why functions in this class are package private and static?

Reply:
ack, changing to non-static and private